config/printer-biqu-bx-2021.cfg

```diff
@@ -0,0 +1,390 @@
+    SET_PIN PIN=screen VALUE=1
+    G28
+    G0 Z1
+    # Warm up nozzle, not to full temps yet
+    M104 S150
+    # Set LED to Purple for bed heating
+    SET_LED LED=led BLUE=0.94 RED=0.63 GREEN=0.13
+    M117 Heating Bed
+
+    # Allow probe to warm up, then re-home Z
+    M190 S60
+    M105
+    G4 P90000
+    G28 Z
+
+    M190 S{params.BED}
+    M105
+    M82
+    G90
+    G21
+    M83
+    G92 E0
+
+    # Set LED to Red for nozzle heating
+    SET_LED LED=led BLUE=0.0 RED=1.0 GREEN=0.0
+    M117 Heating Nozzle
+    G0 X2 Y0 F6000
+    G0 Z0.4
+    M109 S{params.NOZZLE}
+    M105
+    # Set LED to white for printing
+    SET_LED LED=led BLUE=1.0 RED=1.0 GREEN=1.0
+    M117 Printing
+
+    # Purge Line
+    G1 X120 E30 F1200
+    G1 Y1
+    G1 X2 E30 F1200
+    G92 E0
+
+    G1 Z1.0 F600
+    G92 E0
+    G0 F9000
+
+[gcode_macro PRINT_END]
+gcode:
+    M400                             ; wait for buffer to clear
+    M104 S0                          ; turn off hotend
+    M140 S0                          ; turn off bed
+    G92 E0                           ; zero the extruder
+    G1 E-4.0 F3600                   ; retract
+    #   Get Boundaries
+    {% set max_x = printer.configfile.config["stepper_x"]["position_max"]|float %}
+    {% set max_y = printer.configfile.config["stepper_y"]["position_max"]|float %}
+    {% set max_z = printer.configfile.config["stepper_z"]["position_max"]|float %}
+
+    #   Check end position to determine safe directions to move
+    {% if printer.toolhead.position.x < (max_x - 20) %}
+        {% set x_safe = 20.0 %}
+    {% else %}
+        {% set x_safe = -20.0 %}
+    {% endif %}
+
+    {% if printer.toolhead.position.y < (max_y - 20) %}
+        {% set y_safe = 20.0 %}
+    {% else %}
+        {% set y_safe = -20.0 %}
+    {% endif %}
+
+    {% if printer.toolhead.position.z < (max_z - 2) %}
+        {% set z_safe = 2.0 %}
+    {% else %}
+        {% set z_safe = max_z - printer.toolhead.position.z %}
+    {% endif %}
+
+    #  Commence PRINT_END
+    G91                              ; relative positioning
+    G0 Z{z_safe} F3600               ; move nozzle up
+    G0 X{x_safe} Y{y_safe} F20000    ; move nozzle to remove stringing
+
+    M106 S0                          ; turn off fan
+    G90                              ; absolute positioning
+    G0 X{max_x / 2} Y{max_y} F3600   ; park nozzle at rear
+
+
+[gcode_macro CANCEL_PRINT]
+description: Cancel the actual running print
+rename_existing: CANCEL_PRINT_BASE
+gcode:
+  TURN_OFF_HEATERS
+  PRINT_END
+  CANCEL_PRINT_BASE
+
+[gcode_macro M600]
+gcode:
+    {% set X = params.X|default(50)|float %}
+    {% set Y = params.Y|default(0)|float %}
+    {% set Z = params.Z|default(10)|float %}
+    SAVE_GCODE_STATE NAME=PAUSE_state
+    PAUSE
+    G91
+    G1 E-0.8 F2700
+    G0 Z{Z}
+    G90
+    M83
+    G0 X{X} Y{Y} F3000
+
+[gcode_macro RESUME]
+rename_existing: RESUME_BASE
+gcode:
+    RESTORE_GCODE_STATE NAME=PAUSE_state
+    RESUME_BASE
```

Comment:
Macros should only be defined if they are needed to support specific printer functions (https://github.com/Klipper3d/klipper/blob/master/docs/Example_Configs.md 5.iv)

Reply:
The PRINT_START macro is definitely needed to support this printer. This is due to how the ABL is setup and the macro implemented "mimmics" the printers current Marlin behaviour for consistent probing of the bed.

I also setup all the other macros based on user feedback on the stock printer. Should I remove those anyway?

Thanks for the feedback!

Reply:
I don't know this printer, so I guess this is fine. Final call is on Kevin anyway